api/api_errors.go

```diff
@@ -110,3 +118,44 @@ func (errF3ParticipationTicketStartBeforeExisting) Error() string {
 type errF3NotReady struct{}
 
 func (errF3NotReady) Error() string { return "f3 isn't yet ready to participate" }
+
+type ErrExecutionRevertedWithData struct {
+	Message string
+	Data    string
+}
+
+// Error returns the error message.
+func (e *ErrExecutionRevertedWithData) Error() string { return e.Message }
+
+// FromJSONRPCError converts a JSONRPCError to ErrExecutionRevertedWithData.
+func (e *ErrExecutionRevertedWithData) FromJSONRPCError(jerr jsonrpc.JSONRPCError) error {
+	if jerr.Code != EExecutionRevertedWithData {
+		return nil
```

Comment:
let's error in this case - because what will happen here is that the user will get a `ErrExecutionRevertedWithData` error that's completely empty - and this _is_ an error because something got wired up wrong

Reply:
as per my comment in https://github.com/filecoin-project/go-jsonrpc/pull/123 - we should actually handle an error return - reverting to the default message type but complaining about it in logs seems appropriate

Reply:
Done.